OrbitGl/ProcessesDataView.cpp

```diff
@@ -256,38 +208,45 @@ void ProcessesDataView::DoFilter() {
 
 //-----------------------------------------------------------------------------
 void ProcessesDataView::UpdateProcessList() {
-  size_t numProcesses = m_ProcessList.Size();
+  size_t numProcesses = process_list_.size();
   m_Indices.resize(numProcesses);
   for (size_t i = 0; i < numProcesses; ++i) {
     m_Indices[i] = i;
   }
 }
 
 //-----------------------------------------------------------------------------
-void ProcessesDataView::SetRemoteProcessList(ProcessList a_RemoteProcessList) {
-  m_IsRemote = true;
-  m_ProcessList = std::move(a_RemoteProcessList);
+void ProcessesDataView::SetProcessList(
+    const std::vector<std::shared_ptr<Process>>& process_list) {
+  process_list_ = process_list;
   UpdateProcessList();
   OnSort(m_SortingColumn, {});
   OnFilter(m_Filter);
   SetSelectedItem();
 }
 
 //-----------------------------------------------------------------------------
-void ProcessesDataView::SetRemoteProcess(
-    const std::shared_ptr<Process>& a_Process) {
-  std::shared_ptr<Process> targetProcess =
-      m_ProcessList.GetProcess(a_Process->GetID());
-  if (targetProcess) {
-    m_SelectedProcess = a_Process;
-    UpdateModuleDataView(m_SelectedProcess);
+void ProcessesDataView::UpdateProcess(const std::shared_ptr<Process>& process) {
+  auto it =
+      std::find_if(process_list_.begin(), process_list_.end(),
```

Comment:
Do you know if the lambdas always get compiled out?  Seems to me like a for loop in a helper function might be cleaner here, but I don't have a strong opinion on this if the lambdas are compiled out.

Reply:
Apparently they do not... at least according to godbolt.
(I will return to this in one of CL's, going to merge as it for now)

Reply:
Does it matter for code that is not performance-critical?

Reply:
From my previous experience I would expect std::find_if to be more performant than a standard for-loop with a break condition because the raw loop cannot be unrolled as effectively as the std::find_if because the raw-loop body guarantees to be called only until the break occurs while the lambda in std::find_if can be called more often. (It only guarantees to be finished in linear time) And since a shared_ptr deref can have side effects the as-if-rule does not apply here. But I might be wrong about that.